cases: Versioning Data and Model Files (review)

jorgeorpinel opened the pull request:
Addresses #820

- [x] + Simplify the Use Cases index (dvc.org/doc/use-cases)
- [x] Language review + copy edits
- [x] Refocus writing (remove distracting parts).
- [x] Give it a better structure.
- [x] Reference other docs more.
---
- [x] Be very clear and explicit about the role of DVC vs Git
- [x] Expand on run/dag (pipelines) and/or dvc.yaml/lock versioning.
---
- [x] Rewrite as a higher level _Explanation_. See https://github.com/iterative/dvc.org/pull/1716#issuecomment-679316702
- [x] Fix and improve links from other docs to this case and its's tutorial.
- ~~Extract _How-to_ steps into actual how-to guide? (hidden for now, see #1705)~~ Extracted to #1722

jorgeorpinel commented:
Requesting a general review of the doc's structure and direction. Probably easiest to read it here directly: https://dvc-landing-use-cases-iezica0y.herokuapp.com/doc/use-cases/versioning-data-and-model-files.

shcheklein commented:
To be honest I don't see any major changes. What I would expect is that we go at least one level up in terms of how we explain this use case. It is still very technical, with exact commands, etc. We should migrate this into tutorials, how-tos

jorgeorpinel commented:
Agree. My first step was to check that at the current level the doc covers everything it should. Now I'm working on rewriting it to fit in the use case category.

jorgeorpinel commented:
BTW, based on https://documentation.divio.com/introduction/ Use Cases are not fully documentation. They are part _Explanation_ I guess, and they can contain some _How-to_ steps (in fact the Shared Cache one is one of our most popular "tutorials", I think). But they're also 50% marketing, closer to our current https://dvc.org/features page sections.

May be interesting to pull them out of /docs, even merge with /features perhaps. Moving the How-to parts to the actual /how-to docs of course (and link from the use case). This would make them proper landing pages (possibly with their own design, even).

jorgeorpinel commented:
> I would expect is that we go at least one level up in terms of how we explain this use case. It is still very technical

I simplified the intro and **DVC is not Git!** sections of the use case. See https://dvc-landing-use-cases-iezica0y.herokuapp.com/doc/use-cases/versioning-data-and-model-files.

The rest of the content was also updated but it will need to be migrated like you mentioned, and just briefly mentioned//linked to the new location. However that's also the case for all other use cases including [Data Registries](https://dvc.org/doc/use-cases/data-registries), and depends a little on having started the How-to section (in progress, see #899). So can we try to merge this and let me do the same initial simplification to the remaining use cases @shcheklein ?

I'll open a separate issue for migrating the how-to/tutorial parts.

I also improved mentions of the versioning case and it's tutorials throughout the docs.

jorgeorpinel commented:
> So can we try to merge this and let me

Per call with Ivan, we'll use this same PR to try and simplify the 2nd half of the Versioning use case, moving low-level code blocks to its tutorial... ⏳

jorgeorpinel commented:
OK @shcheklein I shortened the more instructional parts of the use case (from **Track data and models for versioning** on). It's still long but much more like a generic demo than a how-to / tutorial. Similar to the level of the **Data Registries** use case. PTAL

Track data and models for versioning

We may need to further shorten these use cases and make/reuse their tutorials instead, but I feel like that should be a separate issue? Thanks

jorgeorpinel commented:
See https://dvc-landing-use-cases-iezica0y.herokuapp.com/doc/use-cases/versioning-data-and-model-files

shcheklein commented on content/docs/command-reference/import.md:
```diff
@@ -65,7 +65,7 @@ path in the <abbr>workspace</abbr>. It records enough metadata about the
 out of date.
 
-To actually [track the data](/doc/tutorials/get-started/data-versioning),
+To actually [version the data](/doc/tutorials/get-started/data-versioning),
```
not sure this is the best term ... but okay, let' keep it for now

jorgeorpinel replied:
"To actually version the data, git add (and git commit) the import stage." seems accurate to me 🙂

jorgeorpinel replied:
Plus the link goes to the Versioning use case.

shcheklein commented on content/docs/command-reference/add.md:
```diff
@@ -19,17 +19,18 @@ The `dvc add` command is analogous to `git add`, in that it makes DVC aware of
 
-> See also `dvc run` for more advanced ways to version intermediate and final
-> results (like ML models).
+> See also `dvc.yaml` and `dvc run` for more advanced ways to track & version
```
& -> and

jorgeorpinel replied:
Updated.

shcheklein commented on content/docs/use-cases/versioning-data-and-model-files/index.md:
```diff
@@ -1,77 +1,121 @@
-file. Let's consider the full checkout first. It's quite straightforward:
+> See also `dvc.lock`.
+
+## Switching versions
```
we don't need 3-4 technical sections, we need to give some sense how does it look like, but it's not "how to".

jorgeorpinel replied:
OK, in order to remove more stuff I need to make sure whatever I remove here is covered by the tutorial so I'll review both now ⏳

jorgeorpinel replied:
OK, it wasn't that difficult. See latest version please. I think this is pretty much the shortest it can be without completely removing the 2 technical sections left now, I think.

shcheklein commented:
@jorgeorpinel looks better, but it is still stays on the same level (it feels this at least), we still need to go level or two up.

jorgeorpinel commented:
Alright. I went just 1 level up now, I think. Not sure anything in the technical sections would be left if we go up 2 levels (depending what you meant by that).

shcheklein commented:
@jorgeorpinel good progress, but overall I feel that we moved 0.5 level, not 1 👍 

What I would expect to see to make my feedback more actionable:

- Why versioning of large files is important at all. In ML or not.
- Why DVC is beneficial for this
- Give some sense how does DVC feel and look, etc. See the philosophy
- Connect with other cases

I guess, my point is that most likely we would need to write a new doc, not edit the existing one, to actually move +1 level.

jorgeorpinel commented:
Understood. I'll work on rewriting it completely then ⏳ 

Should we merge this for now?

jorgeorpinel commented:
Or should we close it in favor of #1747 ?

shcheklein commented:
> Or should we close it in favor of #1747 ?

@jorgeorpinel not sure I can recommend on this to be honest, it depends on your flow. From my end, it would be great to have a single PR.

jorgeorpinel commented:
I think this one is done. Maybe it's not what we need but if it's an improvement and you found no errors, etc. then I would like to merge it. But we can also close it since it will be overwritten by the other PR at some point, hopefully soon.

shcheklein commented:
Fine, we can merge it!

> Maybe it's not what we need but if it's an improvement and you found no errors, etc

I didn't look for errors to be honest, was not reviewing it to deep since to my mind we need to redo this case on a different level.

jorgeorpinel commented:
OK thanks! I think it's good to merge this because people are reading it and it's quite outdated.
